contrib_bots/lib/CommuteBot/CommuteBot.py

```diff
@@ -0,0 +1,150 @@
+
+        return is_commute
+
+    # allows bot to send info via private message or same stream depending on
+    # user input
+    def send_info(self, message, letter, client):
+        if message['type'] == 'private':
+            client.send_message(dict(
+                type='private',
+                to=message['sender_email'],
+                content=letter,
+            ))
+        else:
+            client.send_message(dict(
+                type='stream',
+                subject=message['subject'],
+                to=message['display_recipient'],
+                content=letter,
+            ))
+
+    def handle_message(self, message, client, state_handler):
+        original_content = message['content']
+        content_list = original_content.split()
+
+        if "help" in content_list:
+            help_info = '''
+    Obligatory Inputs:
+        Origin (origin=New York,NY,USA)
+        Destination (destinations=Chicago,IL,USA)
+        Mode Of Transport (Accepted inputs:driving, walking, bicycling, transit) (mode=driving)
+    Optional Inputs:
+        Units (Metric or imperial) (units=metric)
+        Restrictions (Accepted inputs:tolls, highways, ferries, indoor) (avoid=tolls)
+        Departure Time (Seconds After Midnight January 1, 1970 UTC) (departure_time=now)
+        Arrival Time (Seconds After Midnight January 1, 1970 UTC) (arrival_time=732283981023)
+        Language (Available languages available here: https://developers.google.com/maps/faq#languagesupport) (lan=fr)
+
+    Sample request:
+        @commute origin=Chicago,IL,USA destination=New+York,NY,USA mode=driving language=fr
+
+    Please note:
+        Fare information can be derived, though is solely dependent on the
+         availability of the information released by public transport operators
+        Duration in traffic can only be derived if a departure time is set.
+        If a location has spaces in its name, please use a + symbol in the place of the space/s
+        A departure time and a arrival time can not be inputted at the same time
+        To add more than 1 input for a category,
+        eg. more than 1 destinations, use |, eg. destinations=Trump+Tower|The+White+House
+        No spaces within addresses.
+                            '''
+            self.send_info(message, help_info, client)
+        else:
+            para = {}
+            # converts input parameters for api into a dictionary structure for more aethestic coding
+            for item in content_list:
+                # enables key value pair
+                org = item.split('=')
+
+                # ensures that invalid inputs are not entered into url request
+                if len(org) != 2:
+                    continue
+
+                key = org[0]
+                value = org[1]
+                para.update({key: value})
+
+            # adds API Authentication Key to url request
+            para.update({'key': 'AIzaSyCGrx2FIMDVvfjYGgIjN9ocKyd5cfOiH-M'})
```

Comment:
The API key should be stored in a separate file - the bot should be usable without changing its code. Please create a separate file for this and include information in your docs on how to edit it.